src/components/sidenav/sidenav.js

```diff
@@ -323,6 +326,93 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
       }
     }
 
+    function enableDragging() {
+      $mdGesture.register(element, 'drag', { horizontal: true });
+
+      element
+        .on('$md.dragstart', onDragStart)
+        .on('$md.drag', onDrag)
+        .on('$md.dragend', onDragEnd);
+
+      var style = getComputedStyle(element[0]);
+      var sidenavWidth = parseInt(style.width);
+      var isRightSidenav = element.hasClass('md-sidenav-right');
+      var accelerationBound = 10;
+
+      var dragCancelled = false;
+      var dragPercentage;
+      var lastOpenState;
+      var lastDistance = 0;
+      var isQuickDrag = false;
+
+      function onDragStart() {
+        if (element.hasClass('md-locked-open')) {
+          dragCancelled = true;
+        } else {
+          lastOpenState = scope.isOpen;
+          element.css($mdConstant.CSS.TRANSITION_DURATION, '0ms');
+        }
+      }
+
+      function onDrag(ev) {
+        if (dragCancelled) return;
+
+        if (!isQuickDrag) {
+          if (isRightSidenav) isQuickDrag = lastDistance - ev.pointer.distanceX <= -accelerationBound;
+          else isQuickDrag = lastDistance - ev.pointer.distanceX >= accelerationBound;
+        } else {
+          if (isRightSidenav && lastDistance > ev.pointer.distanceX) isQuickDrag = false;
+          else if (!isRightSidenav && lastDistance < ev.pointer.distanceX) isQuickDrag = false;
+        }
+
+        dragPercentage = Math.round((ev.pointer.distanceX / sidenavWidth) * 100);
+        if (!isRightSidenav) dragPercentage = 0 - dragPercentage;
+
+        if (dragPercentage > 100) dragPercentage = 100;
+        else if (dragPercentage < 0) dragPercentage = 0;
+
+        element.css($mdConstant.CSS.TRANSFORM, 'translate3d(-' + (isRightSidenav ? 100 - dragPercentage : dragPercentage) + '%,0,0)');
+        lastDistance = ev.pointer.distanceX;
+      }
+
+      function onDragEnd() {
+        if (dragCancelled) {
+          dragCancelled = false;
+          return;
+        }
+
+        var remainingPercentage = 100 - dragPercentage;
+        var animationTime = 4 * remainingPercentage;
+        var isOpen = dragPercentage > 50 || isQuickDrag;
+        if (isRightSidenav) isOpen = !isOpen;
+
+        element.css($mdConstant.CSS.TRANSITION_DURATION, animationTime + "ms");
+
+        if (isOpen) {
+          element.css($mdConstant.CSS.TRANSFORM, 'translate3d(-100%,0,0)');
+        } else {
+          element.css($mdConstant.CSS.TRANSFORM, 'translate3d(-0%,0,0)');
+        }
+
+        // Reset drag
+        lastDistance = 0;
+        isQuickDrag = false;
+        $timeout(onAnimationDone, animationTime, true, (isRightSidenav ? isOpen : !isOpen));
```

Comment:
Sorry if I don't get your question (really stressed due this days :smile:) - This is only happening on the right Sidenav, because as you see above I invert the `isOpen` value, because the sidenav is aligned at the other side of the page, so I mirror that. And then after I want to complete the Animation I need to get the end value, and I can do this by reverting the value back.